spring-integration-core/src/main/java/org/springframework/integration/dispatcher/PartitionedDispatcher.java

```diff
@@ -151,10 +151,16 @@ public boolean dispatch(Message<?> message) {
 		return partitionDispatcher.dispatch(message);
 	}
 
-	private synchronized void populatedPartitions() {
+	private void populatedPartitions() {
 		if (this.partitions.isEmpty()) {
```

Comment:
You will probably need a `// NOSONAR` here (and other unsynchronized uses) to prevent Sonar from complaining.

Reply:
I'm not sure how that suppose to complain since this property is `final`.
Plus our Sonar instance it out of order for a while.
I'd prefer to look into this when we come back into business with Sonar and it indeed will complain respectively.

Reply:
I don't think being final matters, I believe Sonar will report a smell for inconsistent synchronization of fields.